### PR TITLE
Feature: improve guide on motivation; leverage auto-linking of task; add more fringe options

### PR DESCRIPTION
**Pull request by allanpaiste**
# Motivation

Have template that caters for all possible use-cases when one is creating a PR.

Have template leaning on features enabled for our repositories.

# Description[^1]

In some cases I found myself doing PRs where there were no tests added, but I had validated locally that everything was still working as expected (adding new type, updating local-dev). Decided to add such option here as well.

Additionally I saw some cases where people had provided the JIRA ticket link in non-MD manner. The fact is that we have automatic-linker enabled where mentioning a ticket is enough to turn it into a link.

Also extended the motivation guide to be more demanding and removed the mention of "motivation" from Description (where that would lean more towards what was done based on that motivation).

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Tested manually (previewed the changes)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

[^1]: Here is how to conduct a [code review](https://katanaos.atlassian.net/wiki/spaces/EN/pages/82116626/How+to+conduct+Code+Review)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,13 +15,15 @@ Please delete options that are not relevant.
...
 - [ ] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Maintenance (changes meta-data or package setup)
 
 # How Has This Been Tested?
 
 Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
 
 - [ ] Unit tests
 - [ ] Integration test
```
Comment (FortunatM) [inline .github/pull_request_template.md]: tests

Comment (allanpaiste) [reply .github/pull_request_template.md]: - [x] normalise option namings

**Review by FortunatM — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # Motivation
-A jira link to the [ticket](https://katanaos.atlassian.net/browse/[ticket-number])
+
+Short explanation on why this change is needed and reference to ticket (link created automatically): TICKET-123
```
Comment (FortunatM) [inline .github/pull_request_template.md:3]: @allanpaiste The links are created automatically?

Comment (bogatyrjov1) [reply .github/pull_request_template.md:3]: yes, for every repo the rules are defined in https://github.com/katana-scm/katana-ui/settings/key_links

**Pull request merged**